internal/templates/testing.tmpl

```diff
@@ -0,0 +1,58 @@
+{{if .Tags}}//go:build {{.Tags}}
+
+{{end -}}
+package {{.Package}}
+
+// Code generated by weld. DO NOT EDIT.
+
+import (
+	"github.com/luno/jettison/errors"
+{{range .Imports -}}
+	{{if .Aliased}}{{.Name}} {{end}}"{{.PkgPath}}"
+{{end}}
+)
+
+func NewTesting{{.BackendsName}}(b {{.BackendsType}}) *Testing{{.BackendsName}} {
+	return &Testing{{.BackendsName}}{
+		{{.BackendsType}}: b,
+	}
+}
+
+type Testing{{.BackendsName}} struct {
+	{{.BackendsType}}
+
+{{range .Deps}}{{if not .IsDuplicate -}}
+	{{.Var}} {{.Type}}
+{{end}}{{end}}
+}
+
+{{range .Deps}}
+func (ti *Testing{{$.BackendsName}}) {{.Getter}}() {{.Type}} {
+    if ti.{{.Var}} != nil {
+       return ti.{{.Var}}
+    }
+
+	return ti.Backends.{{.Getter}}()
+}
+{{end}}
+
+{{if .TransBcks -}}
+// Transitive dependency interface assertions.
+var (
+{{range .TransBcks -}}
+	_ {{.}} = (*Testing{{$.BackendsName}})(nil)
+{{end}}
+)
+{{end}}
+
+
+{{range .Deps }}
+// Set{{.Getter}}ForTesting is a runtime available override for the "{{.Var}}" dependency that should only be used for testing.
+func (ti *Testing{{$.BackendsName}}) Set{{.Getter}}ForTesting(t *testing.T, {{.Var}} {{.Type}}) {
```

Comment:
This results "SetGetFooForTesting" because it depends on the naming of the interface method, would it be better to use "Set{{.Type}}ForTesting"? so that you get SetFooForTesting regardless of whether the getter is called GetFoo() or just Foo()

Reply:
Ah good point. Would be a much nicer output. Will update

Reply:
Would the type not also include the package it's from though?

Reply:
As mentioned will update the example and leave as is